#VRML_SIM R2019b utf8
WorldInfo {
}
Viewpoint {
  orientation -0.11700788676506575 -0.6052283333669942 -0.7874057524076035 2.9699382251506172
  position -0.008809534763781879 3.5003174620803 0.7671824035095496
}
TexturedBackground {
}
RectangleArena {
  translation 0.001241 0 -0.183617
  floorTileSize 1 1
  floorAppearance PBRAppearance {
    baseColorMap ImageTexture {
      url [
        "C:/Users/USER/AppData/Local/Programs/Webots/projects/robots/gctronic/e-puck/worlds/textures/floor.png"
      ]
    }
    metalness 0
  }
}
Robot {
  translation -0.065134 0.03 0.066829
  rotation 0 1 0 0.385608
  children [
    LightSensor {
      translation 0.04 0 0.1
      rotation 0 1 0 -1.4399003061004252
      children [
        Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "sensor1"
      rotationStep 0.261799
      lookupTable [
        0 0 0
        1000 1000 0
      ]
    }
    LightSensor {
      translation -0.04 0 0.1
      rotation 0 1 0 -1.7016996938995748
      children [
        Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "sensor2"
      lookupTable [
        0 0 0
        1000 1000 0
      ]
    }
    DEF wheel1 HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.06 0 0.05
      }
      device [
        DEF wheel1 RotationalMotor {
          name "wheel1"
        }
      ]
      endPoint Solid {
        translation 0.06 1.3198722809494515e-05 0.049971517525033836
        rotation -0.05668595823729211 0.05668614910613312 0.9967815119865724 1.5740233667850743
        children [
          DEF wheel Shape {
            appearance PBRAppearance {
              baseColor 0 1 0
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.008
              radius 0.03
            }
          }
        ]
        boundingObject USE wheel
        physics Physics {
        }
        rotationStep 0.261799
      }
    }
    DEF wheel2 HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.06 0 0.05
      }
      device [
        DEF wheel2 RotationalMotor {
          name "wheel2"
        }
      ]
      endPoint Solid {
        translation -0.06 0 0.05
        rotation -1.4239046986508296e-07 1.4239094930950726e-07 0.9999999999999797 1.570799693899595
        children [
          DEF wheel Shape {
            appearance PBRAppearance {
              baseColor 0 1 0
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.008
              radius 0.03
            }
          }
        ]
        name "solid(1)"
        boundingObject USE wheel
        physics Physics {
        }
      }
    }
    DEF wheel3 HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.06 0 -0.05
      }
      device [
        DEF wheel3 RotationalMotor {
          name "wheel3"
        }
      ]
      endPoint Solid {
        translation 0.06 0 -0.05
        rotation -1.4239046986508296e-07 1.4239094930950726e-07 0.9999999999999797 1.570799693899595
        children [
          DEF wheel Shape {
            appearance PBRAppearance {
              baseColor 0 1 0
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.008
              radius 0.03
            }
          }
        ]
        name "solid(2)"
        boundingObject USE wheel
        physics Physics {
        }
      }
    }
    DEF wheel4 HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.06 0 -0.05
      }
      device [
        DEF wheel4 RotationalMotor {
          name "wheel4"
        }
      ]
      endPoint DEF wheel4 Solid {
        translation -0.06 0 -0.05
        rotation -1.4239046986508296e-07 1.4239094930950726e-07 0.9999999999999797 1.570799693899595
        children [
          DEF wheel Shape {
            appearance PBRAppearance {
              baseColor 0 1 0
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.008
              radius 0.03
            }
          }
        ]
        name "solid(3)"
        boundingObject USE wheel
        physics Physics {
        }
      }
    }
    DEF body Shape {
      appearance PBRAppearance {
        baseColor 1 0 0
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.1 0.04 0.2
      }
    }
  ]
  boundingObject USE body
  physics Physics {
    mass 8
  }
  controller "my_controller"
}
Solid {
  translation 0.314245 0.03 -0.320918
  rotation 0 -1 0 0.6715559999999999
  children [
    DEF light Shape {
      appearance PBRAppearance {
        baseColor 1 1 0
        metalness 0
      }
      geometry Sphere {
        radius 0.02
      }
      castShadows FALSE
    }
    PointLight {
      attenuation 0 0 1
      color 1 1 0
      location 0 0.3 0
    }
  ]
}
